src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/RunningJobs.java

```diff
@@ -229,23 +235,29 @@ private void cancelMatchingJobs(GerritTriggeredEvent event, String matchOnJobNam
            // Interrupt any currently running jobs.
            Jenkins jenkins = Jenkins.get();
            for (Computer c : jenkins.getComputers()) {
-               List<Executor> executors = new ArrayList<Executor>();
+               List<Executor> executors = new ArrayList<>();
                executors.addAll(c.getOneOffExecutors());
                executors.addAll(c.getExecutors());
                for (Executor e : executors) {
                    Queue.Executable currentExecutable = e.getCurrentExecutable();
+                   if (!(currentExecutable instanceof Run<?, ?>)) {
+                       continue;
+                   }
+
+                   Run<?, ?> run = (Run<?, ?>)currentExecutable;
+                   if (!checkCausedByGerrit(event, run.getCauses())) {
+                       continue;
+                   }
+
+                   String runningJobName = run.getParent().getFullName();
+                   if (!jobName.equals(runningJobName)) {
+                       continue;
+                   }
 
-                   if (currentExecutable != null && currentExecutable instanceof Run<?, ?>) {
-                       Run<?, ?> run = (Run<?, ?>)currentExecutable;
-                       if (checkCausedByGerrit(event, run.getCauses())) {
-                           if (matchOnJobName == null || matchOnJobName.equals(run.getParent().getFullName())) {
-                               e.interrupt(
-                                       Result.ABORTED,
-                                       new NewPatchSetInterruption()
-                               );
-                           }
-                       }
+                   if (!(this.job instanceof FreeStyleProject)) {
```

Comment:
Need to check on this again, since there could be other job types as well...

Reply:
There are some others it could be, like `MatrixProject`. Probably checking if it is `AbstractProject` would work and cover all of them except `WorkflowJob`.

Reply:
Do you have any clue why `item.getCauses()` does not work for `hudson.model.Queue$BuildableItem:ExecutorStepExecution.PlaceholderTask` queue items? Even if the build is triggered / scheduled via the Plugin the causes are missing at least this is what I got for WorkflowJobs.

Reply:
Just did some tests and as soon as the build is not displayed as "part of xyz" we can actually get the assigned causes. Since I'm not so deep into the whole Jenkins dev yet I try to understand the different behaviors and states in the queue.
However with this additional check we still get leftovers in the queue marked as "Stopping part of  xyz".
I think some additional logic is required to really take care of that. At least for now the job is aborted just once and not multiple times.